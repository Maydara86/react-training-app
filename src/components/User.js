import React from 'react'

const user = props => {
  return(
    <div>
      {/* <img src="{props.image}" alt={props.name} /> */}
      <div>name: {props.name}</div>
      <button onClick={() => props.handelFollowClick(props.id)}>
        {props.isFollowed ? 'Unfollow' : 'Follow'}
      </button>
      <input 
        checked={props.isStared} 
        onChange={() => props.handelStarClick(props.id)}
        type='checkbox' 
      />
      <div>date: {props.date}</div>
      <div>reading time: {props.readingTime}</div>
      <div>image here</div>
      <hr />
    </div>
  )
}

export default user