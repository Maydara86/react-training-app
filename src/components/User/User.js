import React from 'react';
import styles from './User.module.css';
import myImage from '../../assets/images/avatar.png';
import PropTypes from 'prop-types';

const User = ({ id, name, date, isFollowed, isStarred, handleFollowClick, handleStarClick, readingTime, useSimplifiedLayout }) => {
  
  const starColor = isStarred ? 'gold' : 'grey'
    return(
      <div className={useSimplifiedLayout ? styles.SmallUserContainer : styles.UserContainer} key={id}>
        <img className={useSimplifiedLayout ? styles.SmallUserImageItem : styles.UserImageItem} src={myImage} alt={name} />
        <div className={useSimplifiedLayout ? styles.SmallUserInfoItem : styles.UserInfoItem}>
          <div className={useSimplifiedLayout ? styles.SmallContentContainerRow1 : styles.ContentContainerRow1}>
            <div className={styles.NameItem}>{name}</div>
            {!useSimplifiedLayout && <button className={styles.FollowItem} onClick={() => handleFollowClick(id)}>
              {isFollowed ? 'Unfollow' : 'Follow'}
            </button>}
          </div>
          <div className={styles.ContentContainerRow2}>
            <div className={styles.DateItem}>{date}</div>
            <span>Â·</span>
            <div className={styles.ReadingTimeItem}>{readingTime} min read</div>
            <Icon className={styles.StarItem} color={starColor} handleStarClick={handleStarClick} id={id} />
          </div>
        </div>
      </div>
    )

}

export function Icon({color, handleStarClick, id}) {
  return (
    <svg
      style={{marginTop: 4, padding: 4, width: 15, height: 15, fill: color, cursor: 'pointer'}}
      viewBox="0 0 15 15"
      onClick={() => handleStarClick(id)}
    >
      <path d="M7.44 2.32c.03-.1.09-.1.12 0l1.2 3.53a.29.29 0 00.26.2h3.88c.11 0 .13.04.04.1L9.8 8.33a.27.27 0 00-.1.29l1.2 3.53c.03.1-.01.13-.1.07l-3.14-2.18a.3.3 0 00-.32 0L4.2 12.22c-.1.06-.14.03-.1-.07l1.2-3.53a.27.27 0 00-.1-.3L2.06 6.16c-.1-.06-.07-.12.03-.12h3.89a.29.29 0 00.26-.19l1.2-3.52z" />
    </svg>
  );
}

User.propTypes = {
  handleFollowClick: PropTypes.func.isRequired,
  handleStarClick: PropTypes.func.isRequired
};

export default User;
