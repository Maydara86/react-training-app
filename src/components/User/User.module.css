.container {
  /* border: 2px solid #000; */
  margin: 5px;
  /*padding: 5px; */
  display: flex;
  flex-wrap: wrap;
  flex-flow: row wrap;
  align-content: center;
}

.contentContainer {
  display: flex;
  /* border: 2px solid yellow; */
  flex-flow: column wrap;
}

.contentContainerRow1 {
  /* border: 2px solid blue; */
  /* margin: 5px; */
  /* padding: 5px;  */
  display: flex;
  flex-wrap: wrap;
  flex-flow: row wrap;
  height: 50%;
}

.contentContainerRow2 {
  /* border: 2px solid violet; */
  /* margin: 5px;
  padding: 5px; */
  display: flex;
  flex-wrap: nowrap;
  flex-flow: row wrap;
  height: 50%;
}

.imageContainer {
  margin: 5px;
  padding: 5px;
  display: flex;
  border-radius: 100%;
  border: 1px solid red;
  align-self: center;
  flex: 0 1 30px;
}

.contentContainerRow1 > button {
  margin: 0 5px 0 5px;
  flex: 0 1 20px;
  flex-flow: column wrap;
  height: 60%;
  order: 1;
}

.name {
  margin: 0 5px 0 5px;
  height: 50%;
}

.date {
  margin: 0 5px 0 5px;
  height: 50%;
  align-self: flex-end;
  /* justify-content: flex-start; */
}

.time {
  margin: 0 5px 0 5px;
  height: 50%;
  align-self: flex-end;
  /* justify-content: flex-start; */
}


/* .User {
    width: 60%;
    margin: 16px auto;
    border: 1px solid #eee;
    box-shadow: 0 2px 3px #ccc;
    padding: 16px;
    text-align: center;
}

.User .name {
    font-family: Verdana, Geneva, sans-serif;
    color: red;
} */

/* Icon Pop */
/* @-webkit-keyframes hvr-icon-pop {
    50% {
      -webkit-transform: scale(1.5);
      transform: scale(1.5);
    }
  }
  @keyframes hvr-icon-pop {
    50% {
      -webkit-transform: scale(1.5);
      transform: scale(1.5);
    }
  }
  .hvr-icon-pop {
    display: inline-block;
    vertical-align: middle;
    -webkit-transform: perspective(1px) translateZ(0);
    transform: perspective(1px) translateZ(0);
    box-shadow: 0 0 1px rgba(0, 0, 0, 0);
    -webkit-transition-duration: 0.3s;
    transition-duration: 0.3s;
  }
  .hvr-icon-pop .hvr-icon {
    -webkit-transform: translateZ(0);
    transform: translateZ(0);
    -webkit-transition-duration: 0.3s;
    transition-duration: 0.3s;
    -webkit-transition-property: transform;
    transition-property: transform;
    -webkit-transition-timing-function: ease-out;
    transition-timing-function: ease-out;
  }
  .hvr-icon-pop:hover .hvr-icon, .hvr-icon-pop:focus .hvr-icon, .hvr-icon-pop:active .hvr-icon {
    -webkit-animation-name: hvr-icon-pop;
    animation-name: hvr-icon-pop;
    -webkit-animation-duration: 0.3s;
    animation-duration: 0.3s;
    -webkit-animation-timing-function: linear;
    animation-timing-function: linear;
    -webkit-animation-iteration-count: 1;
    animation-iteration-count: 1;
  } */