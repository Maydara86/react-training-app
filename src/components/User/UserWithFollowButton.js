import React from 'react'

const UserWIthFollowButton = () => {
  return <div>HI</div>
}

export default UserWIthFollowButton
