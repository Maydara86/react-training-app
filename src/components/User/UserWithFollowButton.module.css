.userContainer {
    display: -ms-flexbox;
    display: -webkit-flex;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    -webkit-align-content: stretch;
    -ms-flex-line-pack: stretch;
    align-content: stretch;
    -webkit-align-items: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
    height: 100px;
    width: 320px;
}

.userImageItem:nth-child(1) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
    border-radius: 100%;
    height: 90px;
    width: 90px;
}

.userInfoItem:nth-child(2) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
    font-size: 16px;
    font-weight: 300;
}



.contentContainerRow1 {
    display: -ms-flexbox;
    display: -webkit-flex;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-content: stretch;
    -ms-flex-line-pack: stretch;
    align-content: stretch;
    margin-top: 20px;
}

.nameItem:nth-child(1) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
    margin: 0 5px 0 5px;
}

.followItem:nth-child(2) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
    border-color: rgba(0, 0, 0, 0.54);
    font-size: 15px;
    text-decoration: none;
    display: inline-block;
    border-style: solid;
    border-width: 1px;
    border-radius: 4px;
    background: 0;
    letter-spacing: 0px;
    font-style: normal;
    color: rgba(0, 0, 0, 0.84);
    fill: rgba(0, 0, 0, 0.84);
    box-sizing: border-box;
    font-weight: 400;
    overflow: visible;
    font: 400 13.3333px Arial;
    font-family: Arial, Helvetica, sans-serif;
    cursor: pointer;
    transition: border-color 0.5s, color 0.5s, fill 0.5s
}

.followItem:hover {
    border-color: rgba(0, 0, 0);
    color: rgba(0, 0, 0);
    fill: rgba(0, 0, 0);
}


.contentContainerRow2 {
    display: -ms-flexbox;
    display: -webkit-flex;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    -webkit-align-content: stretch;
    -ms-flex-line-pack: stretch;
    align-content: stretch;
    color: rgba(0, 0, 0, 0.54);
    line-height: 30px;
}

.dateItem:nth-child(1) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
    margin: 0 0 0 5px;
}

.readingTimeItem:nth-child(2) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
}

.starItem:nth-child(3) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
}

span {
    margin: 0 5px 0 5px;
}