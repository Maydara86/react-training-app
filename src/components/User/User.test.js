import React from 'react';
import renderer from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
import {shallow, mount} from '../../enzyme';

import User, {Icon} from './User';

const user = {
  "id": "5d552d0058f193f2795fc814",
  "isFollowed": "active",
  "isStared": "idle",
  "image": "./assets/images/avata.png",
  "readingTime": 20,
  "name": "Walton Morton",
  "date": "Aug 9"
}

it('renders correctly when there is one user', () => {

  const tree = renderer.create(<User 
    key={user.id}
    id={user.id}
    name={user.name}
    date={user.date}
    readingTime={user.readingTime}
    isStared={user.isStared}
    isFollowed={user.isFollowed}
    image={user.image}
    handleFollowClick={() => 'test'}
    handleStarClick={() => {}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});



it('when the follow button is clicked a callback is executed', () => {
  const mockFollowClick = jest.fn();

  const tree = renderer.create(<User 
    key={user.id}
    id={user.id}
    name={user.name}
    date={user.date}
    readingTime={user.readingTime}
    isStared={user.isStared}
    isFollowed={user.isFollowed}
    image={user.image}
    handleFollowClick={mockFollowClick}
    handleStarClick={() => {}}
  />)

  const button = tree.root.findByType('button');
  button.props.onClick();
  expect(mockFollowClick).toHaveBeenCalledWith('5d552d0058f193f2795fc814');  
})



it('Change the style of the star svg', () => {
  const mockFollowClick = jest.fn();
  const mockStarClick = jest.fn();

  const testRenderer = TestRenderer.create(<User 
    key={user.id}
    id={user.id}
    name={user.name}
    date={user.date}
    readingTime={user.readingTime}
    isStared={user.isStared}
    isFollowed={user.isFollowed}
    image={user.image}
    handleFollowClick={mockFollowClick}
    handleStarClick={mockStarClick}
  />)

  const testInstance = testRenderer.root
  // const button = testInstance.findByType('button')
  // const icon = testInstance.findByType(Icon)
  // testRenderer.root.findByType(Icon).onClick()
  // testRenderer.root.findByType(Icon).onClick()
  expect(testInstance.findByType(Icon).props.color).toBe('grey')
  testInstance.findByType(Icon).simulate('click')
  expect(testInstance.findByType(Icon).props.color).toBe('gold')


  // button.props.onClick();
  // expect(mockFollowClick).toHaveBeenCalled();

  // button.props.onClick();
  // expect(mockFollowClick).toHaveBeenCalledWith('5d552d0058f193f2795fc814');

  // input.props.onChange();
  // expect(mockStarClick).toHaveBeenCalled();

  // icon.onClick();
  // expect(mockStarClick).toHaveBeenCalledWith('5d552d0058f193f2795fc814');
  
})