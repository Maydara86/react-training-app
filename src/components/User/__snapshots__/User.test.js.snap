// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if the User is called from an article 1`] = `
<div
  className="userContainer"
>
  <img
    alt="Walton Morton"
    className="userImageItem"
    src="avatar.png"
  />
  <div
    className="userInfoItem"
  >
    <div
      className="contentContainerRow1"
    >
      <div
        className="nameItem"
      >
        Walton Morton
      </div>
    </div>
    <div
      className="contentContainerRow2"
    >
      <div
        className="dateItem"
      >
        Aug 9
      </div>
      <span>
        ·
      </span>
      <div
        className="readingTimeItem"
      >
        20
         min read
      </div>
      <svg
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "fill": "grey",
            "height": 15,
            "marginTop": 4,
            "padding": 4,
            "width": 15,
          }
        }
        viewBox="0 0 15 15"
      >
        <path
          d="M7.44 2.32c.03-.1.09-.1.12 0l1.2 3.53a.29.29 0 00.26.2h3.88c.11 0 .13.04.04.1L9.8 8.33a.27.27 0 00-.1.29l1.2 3.53c.03.1-.01.13-.1.07l-3.14-2.18a.3.3 0 00-.32 0L4.2 12.22c-.1.06-.14.03-.1-.07l1.2-3.53a.27.27 0 00-.1-.3L2.06 6.16c-.1-.06-.07-.12.03-.12h3.89a.29.29 0 00.26-.19l1.2-3.52z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`renders correctly when there is one user 1`] = `
<div
  className="userContainer"
>
  <img
    alt="Walton Morton"
    className="userImageItem"
    src="avatar.png"
  />
  <div
    className="userInfoItem"
  >
    <div
      className="contentContainerRow1"
    >
      <div
        className="nameItem"
      >
        Walton Morton
      </div>
    </div>
    <div
      className="contentContainerRow2"
    >
      <div
        className="dateItem"
      >
        Aug 9
      </div>
      <span>
        ·
      </span>
      <div
        className="readingTimeItem"
      >
        20
         min read
      </div>
      <svg
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "fill": "grey",
            "height": 15,
            "marginTop": 4,
            "padding": 4,
            "width": 15,
          }
        }
        viewBox="0 0 15 15"
      >
        <path
          d="M7.44 2.32c.03-.1.09-.1.12 0l1.2 3.53a.29.29 0 00.26.2h3.88c.11 0 .13.04.04.1L9.8 8.33a.27.27 0 00-.1.29l1.2 3.53c.03.1-.01.13-.1.07l-3.14-2.18a.3.3 0 00-.32 0L4.2 12.22c-.1.06-.14.03-.1-.07l1.2-3.53a.27.27 0 00-.1-.3L2.06 6.16c-.1-.06-.07-.12.03-.12h3.89a.29.29 0 00.26-.19l1.2-3.52z"
        />
      </svg>
    </div>
  </div>
</div>
`;
